tests/Storage.test.ts

```diff
@@ -293,4 +293,33 @@ describe('LedgerStorage', () =>
                 });
         });
     });
+
+    it('DB transaction test when writing a block', () =>
+    {
+        let ledger_storage = new LedgerStorage(":memory:", async (err1: Error | null) =>
+        {
+            assert.ok(!err1, err1?.message);
+
+            let block = new Block();
+            block.parseJSON(sample_data[0]);
+
+            await ledger_storage.putEnrollments(block);
+            assert.rejects(ledger_storage.putBlocks(sample_data[0]),
```

Comment:
```suggestion
            await assert.rejects(ledger_storage.putBlocks(sample_data[0]),
```
There's another test left. `await` is required.